cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -231,11 +233,45 @@ protected void startUp() throws Exception {
   /**
    * Initializes the handlers.
    */
-  protected void initHandlers() throws Exception {
-    authenticationHandler.init();
+  private void initHandlers() throws Exception {
+    Map<String, String> handlerProps = getAuthHandlerConfigs(cConfiguration);
+    Map<String, String> secureProps = getAuthHandlerConfigs(sConfiguration);
+
+    // used by CertificateAuthenticationHandler (see CDAP-7287)
+    copyProperty(secureProps, cConfiguration, "security.auth.server.ssl.truststore.path");
+    copyProperty(secureProps, cConfiguration, "security.auth.server.ssl.truststore.type");
+    copyProperty(secureProps, cConfiguration, "security.auth.server.ssl.truststore.password");
+    // used by AbstractAuthenticationHandler
```

Comment:
Maybe we can pass in all of cConf, and only filter sConf. That will simplify some of this code.